src/main/java/seedu/dietbook/calculator/CalculatorData.java

```diff
@@ -0,0 +1,133 @@
+package seedu.dietbook.calculator;
+
+import seedu.dietbook.food.Food;
+import seedu.dietbook.list.FoodList;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CalculatorData {
+    FoodList list;
+    List<Food> foods = null;
+
+    public CalculatorData() {
+        this.list = new FoodList();
+    }
+
+    public CalculatorData(FoodList foodList) {
+        this.list = foodList;
+    }
+
+    public void inputData(FoodList foodList) {
+        this.list = foodList;
+    }
+
+    public List<Integer> getTotalCalorie() {
+        this.foods = list.getFoods();
+        List<Integer> calories = new ArrayList<>();
+        for (Food food : foods) {
+            calories.add(food.getCalorie());
+        }
+        return calories;
+    }
+
+    public List<Integer> getTotalCalorie(LocalDateTime startTime) {
+        this.foods = list.getFoodsAfterDateTime(startTime);
```

Comment:
Similar to the other comment. I suspect that this should be `getPortionedFoodsAfterDateTime()`.

These apply throughout similar methods in this class.